v2/devdoc/log_sink_etw_requirements.md

```diff
@@ -20,160 +20,184 @@ The signature of `log_sink_etw.log_sink_log` is:
 
-`log_sink_etw.log_sink_log` shall call `log_context_get_property_value_pair_count` to obtain the count of properties that are to be added to the ETW event.
+  - **SRS_LOG_SINK_ETW_01_016: [** If `log_level` is `LOG_LEVEL_VERBOSE` the event name shall be `LogVerbose`. **]**
 
-`log_sink_etw.log_sink_log` shall call `log_context_get_property_value_pairs` to obtain the properties that are to be added to the ETW event.
+  - **SRS_LOG_SINK_ETW_01_017: [** Otherwise the event name shall be `Unknown`. **]**
 
-`log_sink_etw.log_sink_log` shall limit the number of properties that are emitted in the ETW event to 64.
+- **SRS_LOG_SINK_ETW_01_018: [** Logging level: **]**
 
-`log_sink_etw.log_sink_log` shall compute the metadata size for the self described event metadata as follows:
+  - **SRS_LOG_SINK_ETW_01_019: [** If `log_level` is `LOG_LEVEL_CRITICAL` the ETW logging level shall be `TRACE_LEVEL_CRITICAL`. **]**
 
-- Length of the event name (excluding zero terminator) + 1.
+  - **SRS_LOG_SINK_ETW_01_020: [** If `log_level` is `LOG_LEVEL_ERROR` the ETW logging level shall be `TRACE_LEVEL_ERROR`. **]**
 
-- Length of the `content` field name + 1.
+  - **SRS_LOG_SINK_ETW_01_021: [** If `log_level` is `LOG_LEVEL_WARNING` the ETW logging level shall be `TRACE_LEVEL_WARNING`. **]**
 
-- Length of the `file` field name + 1.
+  - **SRS_LOG_SINK_ETW_01_022: [** If `log_level` is `LOG_LEVEL_INFO` the ETW logging level shall be `TRACE_LEVEL_INFO`. **]**
 
-- Length of the `func` field name + 1.
+  - **SRS_LOG_SINK_ETW_01_023: [** If `log_level` is `LOG_LEVEL_VERBOSE` the ETW logging level shall be `TRACE_LEVEL_VERBOSE`. **]**
 
-- Length of the `line` field name + 1.
+  - **SRS_LOG_SINK_ETW_01_024: [** Otherwise the ETW logging level shall be `TRACE_LEVEL_NONE`. **]**
 
-- For each property in `log_context`, the length of the property name + 1 and one extra byte for the type of the field.
+- **SRS_LOG_SINK_ETW_01_025: [** If `log_context` is `NULL` only the fields `content`, `file`, `func` and `line` shall be added to the ETW event. **]**
 
-- For struct properties one extra byte shall be added for the field count.
+- **SRS_LOG_SINK_ETW_01_048: [** If `log_context` is not `NULL`: **]**
 
-`log_sink_etw.log_sink_log` shall fill a `SELF_DESCRIBED_EVENT` structure, setting the following fields:
+  - **SRS_LOG_SINK_ETW_01_049: [** `log_sink_etw.log_sink_log` shall call `log_context_get_property_value_pair_count` to obtain the count of properties that are to be added to the ETW event. **]**
 
-- `_tlgBlobTyp` shall be set to `_TlgBlobEvent4`.
+  - **SRS_LOG_SINK_ETW_01_050: [** `log_sink_etw.log_sink_log` shall call `log_context_get_property_value_pairs` to obtain the properties that are to be added to the ETW event. **]**
 
-- `_tlgChannel` shall be set to 11.
+  - **SRS_LOG_SINK_ETW_01_082: [** If more than 64 properties are given in `log_context`, `log_sink_etw.log_sink_log` shall not add any properties to the event. **]**
```

Comment:
>shall not add any [](http://example.com/codeflow?start=118&length=17)

hm. 

Consider adding "some" is better than none at all?

Reply:
Considered and added a task

The problems are with ensuring we keep entire structures. Can be done of course, but would make things somewhat more complex.

Added a separate task for it.
ID Work Item Type Title State Assigned To Effort Remaining Work Original Estimate Value Area Iteration Path Tags
22152827 Task Consider keeping as many as possible of the properties in an event To Do Dan Cristoloveanu         One